openprescribing/gcutils/bigquery.py

```diff
@@ -245,19 +245,25 @@ def upload_model(self, model, table_id=None):
                 table_id = table_id.split('_', 1)[1]
         schema = build_schema_from_model(model)
         table = self.get_or_create_table(table_id, schema)
+        # We reload the schema here, as when older BQ tables were created,
+        # Django DateFields were mapped to BQ TIMESTAMP fields.  However, we
+        # now map them to DATE fields.
+        schema = table.gcbq_table.schema
+
         columns = [
             f.db_column or f.attname
             for f in model._meta.fields
             if not f.auto_created
         ]
+
         timestamp_ixs = [
             ix
             for ix, field in enumerate(schema)
             if field.field_type == 'TIMESTAMP'
         ]
 
         def transformer(record):
-            for ix in timestamp_ixs:
+            for ix in timestamp_ixs and record[ix]:
```

Comment:
:eyes: what is this witchcraft?!  (not sure I'm objecting, maybe it's a common idiom I've just never seen!)

Reply:
Nope, it's a bug -- Python's doing

```
for ix in (timestamp_ixs and record[ix]):
    ...
```

which is equivalent to

```
for ix in timestamp_ixs:
    ...
```

because the `and` is shortcircuiting.

It should be:

```
for ix in timestamp_ixs:
    if record[ix]:
        ...
```

Well spotted, thanks.